cookbook/core/flyte_basics/task_cache.py

```diff
@@ -92,3 +92,105 @@ def square(n: int) -> int:
 #
 # .. note::
 #   The format used by the store is opaque and not meant to be inspectable.
+#
+# Caching of non-Flyte offloaded objects
+# ######################################
+#
+# The behavior displayed by the cache, in some cases, does not match the users intuitions. For example, this code makes use of pandas dataframes:
+
+# .. code-block:: python
+#
+#    @task
+#    def foo(a: int, b: str) -> pd.DataFrame:
+#      df = pd.Dataframe(...)
+#      ...
+#      return df
+#
+#   @task(cached=True, version="1.0")
+#   def bar(df: pd.Dataframe) -> int:
+#       ...
+#
+#   @workflow
+#   def wf(a: int, b: str):
+#       df = foo(a=a, b=b)
+#       v = bar(df=df)
+#
+#
+# One would expect that ``bar`` would be cacheable, but that is not the case due to the representation of dataframes in the Flyte type system. However, starting on flyte release 0.19.3, we provide a way to override the representation of certain objects, including pandas dataframes and other structured datasets. This is done via annotations on the objects, for example, in order to cache the result of calls to ``bar`` we can rewrite the code above like this:
```

Comment:
```suggestion
# If run twice with the same inputs, one would expect that ``bar`` would trigger a cache hit, but it turns out that's not the case because of how dataframes are represented in Flyte. (something here about offloading/pointers?)  However, with release 0.19.3, Flyte provides a new way to control memoization behavior of  `FlyteSchema` and `StructuredDataset` literals. This is done via a `typing.Annotated` call on the task signature.  For example, in order to cache the result of calls to ``bar``, we can rewrite the code above like this:
```